### PR TITLE
Support dlpack for privateuse1

### LINKED ISSUE #129652: Support dlpack for device privateuseone
### 🚀 The feature, motivation and pitch

Currently, dlpack-related functions in PyTorch like _getDLDevice()_ do not include device privateuseone, so custom backend users can not support dlpack on some non-CUDA devices. Could you support this feature in the future? I appreciate any information.

### Alternatives

_No response_

### Additional context

_No response_

cc @bdhirsh @NmomoN @mengpenghui @fwenguang @cdzhan @1274085042 @PHLens

### PR DESCRIPTION
**Pull request by hipudding**
Fixes #129652

### CODE REVIEW

**Review by shink — APPROVED** [review]

File: test/test_cpp_extensions_open_device_registration.py [hunk 1]
```diff
@@ -579,6 +579,13 @@ def test_open_device_numpy_serialization(self):
                     with torch.serialization.skip_data():
                         torch.save(sd, f)
 
+    def test_open_device_dlpack(self):
+        t = torch.randn(2,3).to("foo")
+        capsule = torch.utils.dlpack.to_dlpack(t)
+        t1 = torch.from_dlpack(capsule)
```
Comment (FFFrog) [inline test/test_cpp_extensions_open_device_registration.py:585]:
```suggestion
        t1 = torch.from_dlpack(capsule)
        t1 = torch.from_dlpack(capsule)
        self.assertTrue(t1.device == t.device)
```

**Review by fffrog — APPROVED** [review]

**Comment by hipudding** [comment]
@albanD , Would you please review this PR? Thanks.

**Comment by hipudding** [comment]
Hey @ezyang, Could you please review this PR? Thanks.

**Review by ezyang — CHANGES_REQUESTED** [review]
You can't just add a new value to the dlpack header, it's a standard. Fortunately, I think you just want to use kDLExtDev here

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**